doc/sourmash-internals.md

```diff
@@ -0,0 +1,638 @@
+# A guide to the internals of sourmash
+
+```{contents} Contents
+:depth: 3
+```
+
+sourmash was created in 2015, and has been repeatedly reorganized,
+refactored, and optimized to support ever larger databases, faster
+queries, and new use cases. We've also regularly added new
+functionality and features.  So sourmash can be pretty complicated
+internally, and our user-facing documentation only covers a fraction
+of its potential!
+
+This document is a brain dump intended for expert users and sourmash
+developers who want to understand how, why, and when to use various
+sourmash features. It is unlikely ever to be comprehensive, so the
+information you are interested in may not yet exist in this document,
+but we are always happy to add to it -
+[just ask in an issue!](https://github.com/sourmash-bio/sourmash/issues)
+
+## Signatures and sketches
+
+sourmash operates on sketches. Each sketch is a collection of hashes.
+Each sketch is contained in a signature.
+
+Internally, sketches (class `MinHash`) contain the following information:
+* a set of hashes;
+* an optional abundance for each hash (when `track_abund` is True);
+* a seed;
+* a k-mer size;
+* a molecule type;
+* either a `num` (for MinHash) or a `scaled` value (for FracMinHash);
+
+Signature objects (class `SourmashSignature`) contain a sketch (property `.minhash`) as well as additional information:
+* an optional `name`
+* an optional `filename`
+* a license (currently must be CC0);
+* an `md5sum(...)` method that returns a hash of the sketch.
+
+For now, we tend to refer to signatures and sketches interchangeably,
+because they are almost entirely 1:1 in the code base (but see [sourmash#616](https://github.com/sourmash-bio/sourmash/issues/616)).
+
+The default signature interchange/serialization format is JSON, optionally
+gzipped. This format is written and read by Rust code.
+
+In general, a lot of effort in sourmash is spent managing collections of
+signatures _before_ actually doing comparisons with them; see manifests,
+and `Index` objects, below.
+
+### Making sketches
+
+Sketches are produced by hashing k-mers with murmurhash and then
+keeping either the lowest `num` hashes (for MinHashes sketches) or
+keeping all hashes below `2**64 / scaled` (for FracMinHash sketches).
+
+The default MinHash sketches use parameters so that they are
+compatible with mash sketches.
+
+See [utils/compute-dna-mh-another-way.py](https://github.com/sourmash-bio/sourmash/blob/latest/utils/compute-dna-mh-another-way.py) for details on how k-mers are
+hashed.
+
+Note that if hashes are produced some other way (with a different hash
+function) or from some source other than DNA, sourmash can still work
+with them; only `sourmash sketch` does input checking.
+
+### Compatibility checking
+
+The point of the signatures and sketches is to enable certain kinds of
+rapid comparisons - Jaccard similarity and overlap analysis,
```

Comment:
```suggestion
rapid comparisons - Jaccard similarity, $JS(A,B) = \frac{|A \bigcap B|}{|A \bigcup B|}$ and overlap analysis,
```

Is overlap analysis the overlap coefficient, $OC(A,B) = \frac{|A \bigcap B|}{min(|A|,|B|)}$, or jaccard containment, $JC(A,B) = \frac{|A \bigcap B|}{|A|}$ .

Reply:
changed to:

>Jaccard similarity and number of overlapping k-mers,

which is kind of redundant (you can use # of overlapping k-mers to calculate Jaccard) but I'm ok with it since people are likely to have heard of Jaccard so it bridges to a concept they know.